api/python/quilt3/packages.py

```diff
@@ -486,37 +479,9 @@ def resolve_hash(cls, registry, hash_prefix):
             registry(string): location of registry
             hash_prefix(string): hash prefix with length between 6 and 64 characters
         """
+        # TODO: add pkg_name parameter or simply deprecate?
         assert isinstance(registry, PhysicalKey)
-        assert not str(registry).rstrip('/').endswith('.quilt')
-        if len(hash_prefix) == 64:
-            top_hash = hash_prefix
-        elif 6 <= len(hash_prefix) < 64:
-            matching_hashes = [h for h, _
-                               in list_url(registry.join('.quilt/packages/'))
-                               if h.startswith(hash_prefix)]
-            if not matching_hashes:
-                raise QuiltException("Found zero matches for %r" % hash_prefix)
-            elif len(matching_hashes) > 1:
-                raise QuiltException("Found multiple matches: %r" % hash_prefix)
-            else:
-                top_hash = matching_hashes[0]
-        else:
-            raise QuiltException("Invalid hash: %r" % hash_prefix)
-        return top_hash
-
-    @classmethod
-    def _shorten_tophash(cls, package_name, registry: PhysicalKey, top_hash):
-        min_shorthash_len = 7
-
-        matches = [h for h, _ in list_url(registry.join('.quilt/packages/'))
-                   if h.startswith(top_hash[:min_shorthash_len])]
-        if len(matches) == 0:
-            raise ValueError(f"Tophash {top_hash} was not found in registry {registry}")
-        for prefix_length in range(min_shorthash_len, 64):
-            potential_shorthash = top_hash[:prefix_length]
-            matches = [h for h in matches if h.startswith(potential_shorthash)]
-            if len(matches) == 1:
-                return potential_shorthash
+        return get_package_registry(registry).resolve_top_hash(None, hash_prefix)
```

Comment:
Does this work?

Reply:
Well, yes:
https://github.com/quiltdata/quilt/pull/1682/files#diff-60b178a9c2fe184f8e58ee3927efc3ecR105-R106